File: pladder-domain/src/main/java/com/api/pladder/domain/entity/user/enums/UserType.java
package com.api.pladder.domain.entity.user.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum UserType {
    ADMIN("admin", "/api/admin"),
    DETECTIVE("detective", "/api/detective"),
    CUSTOMER("customer", "/api/customer"),
    UNKNOWN("open", "/api/open");

    private final String authorization;
    private final String requestMapper;
    public static UserType fromString(String value) {
        for (UserType userType : UserType.values()) {
            if (userType.name().equalsIgnoreCase(value)) {
                return userType;
            }
        }
        throw new IllegalArgumentException("존재하지 않는 유저타입 입니다: " + value);
    }
}

File: pladder-domain/src/main/java/com/api/pladder/domain/entity/user/enums/UserType.kt
package com.api.pladder.domain.entity.user.enums

enum class UserType(
    val authorization : String,
    val requestMapper : String,
) {
    ADMIN(authorization = "admin", requestMapper = "/api/admin"),
    DETECTIVE(authorization = "detective", requestMapper = "/api/detective"),
    CUSTOMER(authorization = "customer", requestMapper = "/api/customer"),
    UNKNOWN(authorization = "open", requestMapper = "/api/open");
    companion object {
        fun fromString(value: String): UserType {
            return requireNotNull(entries.find { it.name.equals(value, ignoreCase = true) }) {
                "존재하지 않는 유저타입 입니다: $value"
            }
        }
    }
}